jormungandr/src/fragment/process.rs

```diff
@@ -129,25 +130,32 @@ impl Process {
                                     // This interface only makes sense for messages coming from arbitrary users (like transaction, certificates),
                                     // for other message we don't want to receive them through this interface, and possibly
                                     // put them in another pool.
-
-                                    let stats_counter = stats_counter.clone();
-
-                                    let summary = pool
-                            .insert_and_propagate_all(origin, fragments, fail_fast)
-                            .await?;
-
-                        stats_counter.add_tx_recv_cnt(summary.accepted.len());
-
-                        reply_handle.reply_ok(summary);
+                                    let span = debug_span!("incoming_fragments");
+                                    async {
```

Comment:
Same thing with `.instrument()` here and onward.

Reply:
same as above, we cannot use `span.enter()` in async code